app/controllers/authentication.rb

```diff
@@ -0,0 +1,38 @@
+# error 301 do
+  
+# end
+
+
+before '/*' do 
+  authorize_or_redirect unless request.path_info == '/login' or request.path_info == '/logout'
+end
+
+get '/' do
+ redirect "/login"
+end
+
+get '/login' do
+  redirect "/users/#{session[:user_id]}" if session[:user_id] # redirect_unless_logged_in @TODO
+  erb :login
+end
+
+post '/login' do
+
+  user = User.find_by(handle: params[:handle])
+
+  redirect '/login' unless user
```

Comment:
When authorization failes; you should render the login form again in this request with an error. Dont redirect.